### ISSUE TITLE
'View Yaml' action from list views fails for users who can list but not get a resource

### ISSUE THREAD

**Issue by mantis-toboggan-md** [rancher/dashboard#8606]
To reproduce:
1. Create a custom cluster to role to list a resource
2. Create a standard user
3. Explore a cluster, navigate to 'Cluster & Project Members'
4. Add the standard user to the cluster with the custom role created in step 1
5. Login as the standard user and navigate to the relevant resource's list view
6. Use the kebab menu to select 'view as yaml' for one of the resources listed
Result: The yaml doesn't load and an error is seen in the console:
```
console.js:31 TypeError: Cannot read properties of null (reading 'doneOverride')
    at f.<anonymous> (index.vue?46a7:1:1374)
    at t._render (vue.runtime.esm.js:3569:22)
    at f.r (vue.runtime.esm.js:4081:21)
    at Tn.get (vue.runtime.esm.js:4495:25)
    at Tn.run (vue.runtime.esm.js:4570:22)
    at Sn (vue.runtime.esm.js:4326:13)
    at Array.<anonymous> (vue.runtime.esm.js:1989:12)
    at ce (vue.runtime.esm.js:1915:12)
```
Expected result:
The user can view the yaml of the resource

It's worth noting that 'view config' still works in this scenario. This is probably because when we go from a list view to 'view config' we use data already loaded from the request to find all of that resource, but when we 'view yaml' we re-query the api for that specific resource in yaml form.

**Comment by gaktive** [comment]
Pushing to Q3.

**Comment by cnotv** [comment]
Example configuration with screenshots:

<img width="1912" alt="Screenshot 2023-06-27 at 19 58 19" src="https://github.com/rancher/dashboard/assets/5009481/a37f1296-bf2d-46da-bb7b-e3df5009bdb7">
<img width="1648" alt="Screenshot 2023-06-27 at 19 58 37" src="https://github.com/rancher/dashboard/assets/5009481/ab476cbc-1c70-42ba-9c88-331e1b892a2f">
<img width="1472" alt="Screenshot 2023-06-27 at 19 58 43" src="https://github.com/rancher/dashboard/assets/5009481/8817d65c-0dfc-45ba-8667-ada68c4ff130">
<img width="1538" alt="Screenshot 2023-06-27 at 19 59 59" src="https://github.com/rancher/dashboard/assets/5009481/1e6105b1-ba44-4161-9601-a55da766165c">

**Comment by cnotv** [comment]
Another point which is worth mention, it's possible to access the configuration and switch to YAML, but it would not show anything in it.

![Screenshot 2023-06-28 at 18 40 37](https://github.com/rancher/dashboard/assets/5009481/a7a8be89-577e-4a82-8b39-ae9cf5539409)
![Screenshot 2023-06-28 at 18 40 48](https://github.com/rancher/dashboard/assets/5009481/96aeff73-b10b-4c25-a9f0-f6dc4427e4a0)

**Comment by cnotv** [comment]
@mantis-toboggan-md I think we will need to remove the possibility to navigate in the configuration or retrieve the values in somehow.
When I refresh the page I get this error:
![Screenshot 2023-06-30 at 14 23 08](https://github.com/rancher/dashboard/assets/5009481/c9a4f28d-c348-4581-aa1f-a85cc1da1e9e)

**Comment by cnotv** [comment]
Based on our logic, the ability to link to configuration and YAML editor are based on the presence of links, which in this case are present and allow to view.

![Screenshot 2023-07-03 at 16 38 32](https://github.com/rancher/dashboard/assets/5009481/b9ef3c54-7164-4958-80df-a2d6ffde9441)

Although when I try to open the defined URL that's forbidden 🤔 

![Screenshot 2023-07-03 at 16 40 06](https://github.com/rancher/dashboard/assets/5009481/bebd708e-9539-436b-bb27-1dd0f9050cfc)

Something I can do as Admin

![Screenshot 2023-07-03 at 16 41 11](https://github.com/rancher/dashboard/assets/5009481/5569e7b2-6880-44c0-81c1-73f20839ef94)

**Comment by cnotv** [comment]
As discussed and verified with @richard-cox and @axeal, the schema returns the GET as `resourceMethod`, although it returns 403 when trying to load the resource itself. In the example using a Pod.

![image](https://github.com/rancher/dashboard/assets/5009481/df97dae5-90f1-4671-b1b1-44aca4c66218)
![image](https://github.com/rancher/dashboard/assets/5009481/333f6aa6-a7e7-4db1-998d-0993f01f9a92)

![image](https://github.com/rancher/dashboard/assets/5009481/b4864be2-5b6a-4899-b944-84798b0ff6c6)

The resource is correctly listed:

![image](https://github.com/rancher/dashboard/assets/5009481/37001dfe-e621-48db-bdf0-da653778e7bb)

**Comment by richard-cox** [comment]
**We need someone from the auth team to confirm these two points**
- Users who have permission to `list` a resource do NOT automatically get the `get` permission. This needs to be explicitly set alongside `list`
  - This was partially confirmed for the kube api. Is this true? Does it extend to steve?
- We support the use case where role template gives a resource the `list` permission without the `get` permission. Is this illogical or are there geniune use cases?

To resolve this specific issue the UI needs to confirm the user can `get` a resource. Without working this out via roles, we assume the best way is via the resources schema's resourceMethods `GET` value, or specifically via the resources link.self value (if the resource was fetched via a list request).

There's two bugs
- user has `list` and not `get` permission on pods
- k8s/clusters/c-m-bpwh5cmd/v1/schemas/pod 
    - links array only contains `collection` and `self`
    - resourceMethods contains `GET` <---- bug
- k8s/clusters/c-m-bpwh5cmd/v1/pods
    - succeeds
    - entries contain links `self` and `view` <--- bug (though one could argue the view is there given the kube api endpoint)
- pod's `self` link `k8s/clusters/c-m-bpwh5cmd/v1/pods/<ns>/<pod id>`
    - fails with 403

**We need someone from the auth team to confirm the two bugs**
- User should NOT be able to see the steve schema's resourceMethods GET value if the user does not have GET permissions
- The user should not see the `links.self` value if the user does not have GET permissions (resource fetched via LIST permission)

**Comment by MbolotSuse** [comment]
@richard-cox 

Long story short, there's a steve issue here and I would recommend opening an issue in [rancher/rancher](https://github.com/rancher/rancher) so that we can address it. To respond to your specific questions:

> Users who have permission to list a resource do NOT automatically get the get permission. This needs to be explicitly set alongside list

This is true, and extends to steve as well.

> We support the use case where role template gives a resource the list permission without the get permission. Is this illogical or are there geniune use cases?

Near as I can tell, there are very limited use cases for this. The only thing that I could come up with is using `resourceNames` along with `list` - this will require users to use a `fieldSelector` in the list request (so in theory you could give users `get` access on a few resources and `list` access on a few resources, and this method would at least give those users some ability to use a list command rather than requiring a raw `get` every time).  That being said, my personal opinion would be that we need to stick with upstream here, diverging from them and giving users "get" automatically when they have "list" could lead to bad issues. 

> We need someone from the auth team to confirm the two bugs

> User should NOT be able to see the steve schema's resourceMethods GET value if the user does not have GET permissions

My answer here is subject to some conditions (specifically surrounding the schema's inability to filter by namespace/resources names as seen in rancher/rancher#34731 among others), but generally speaking you are correct here - resourceMethods should not include `GET` if the user can only 'list'. I'd recommend opening up an issue (as recommended above) so the backend can triage this.

> The user should not see the links.self value if the user does not have GET permissions (resource fetched via LIST permission)

This I cannot confirm. Near as I can tell, the `self` value is included in any object. So if a user fetches the resources through list, the entries in that list will contain a `self` link. I would not recommend using this value as a source of truth on user's get permissions on an object, and I don't anticipate us making changes to this value to make it work that way.

In short,  I would not consider this second behavior a bug.

**Comment by gaktive** [comment]
Pushing to Q4 since backend still needs to work on this. Not sure if https://github.com/rancher/rancher/issues/34731 or a new ticket is for UI tracking on the blocked status but I know both @richard-cox and @MbolotSuse are on this. If someone else has to work on this though, let's make it clear where the backend work is tracked.

**Comment by richard-cox** [comment]
Investigated if https://github.com/rancher/rancher/issues/45827 would resolve, unfortunately not. From a quick review

- schema resourceMethods GET should not be populated for resources with only LIST
  - frameworks bug. requires rancher/rancher issue
- ui needs to not use a resource's link.self url to determine GET permission on the resource
  - frameworks bug/enhancement. functionality missing

**Comment by richard-cox** [comment]
linked to https://github.com/rancher/rancher/issues/34731, now unblocked

Think the actual issue / solution is from https://github.com/rancher/rancher/issues/50237#issuecomment-3018192345
